packages/didcomm/src/core/__tests__/PackAndUnpack.test.ts

```diff
@@ -0,0 +1,175 @@
+}
+
+describe('DIDComm message pack and unpack', () => {
+  let didResolver: ExampleDIDResolver;
+  let secretsResolver: ExampleSecretsResolver;
+
+  beforeEach(() => {
+    didResolver = createDidResolver([ALICE_DID_DOC, BOB_DID_DOC]);
+    secretsResolver = createSecretsResolver(ALICE_SECRETS);
+  });
+
+  it('should pack, encrypt, and then unpack a message successfully', async () => {
+    const msg = new Message({
+      id: MESSAGE_ID,
+      typ: MESSAGE_TYP,
+      type: MESSAGE_TYPE,
+      from: ALICE_DID,
+      to: [BOB_DID],
+      created_time: 1516269022,
+      expires_time: 1516385931,
+      body: MESSAGE_BODY,
+    });
+
+    // Pack and encrypt the message
+    const [encryptedMsg, encryptMetadata] = await msg.pack_encrypted(
+      BOB_DID,
+      ALICE_DID,
+      ALICE_DID,
+      didResolver,
+      secretsResolver,
+      {
+        forward: false,
+      },
+    );
+
+    expect(encryptedMsg).toBeDefined();
+    expect(encryptMetadata).toMatchObject({
+      from_kid: ALICE_KID,
+      to_kids: expect.any(Array),
+    });
+
+    // Reinitialize resolvers for the receiving party
+    didResolver = createDidResolver([ALICE_DID_DOC, BOB_DID_DOC]);
+    secretsResolver = createSecretsResolver(BOB_SECRETS);
+
+    // Unpack and decrypt the message
+    const [unpackedMsg, unpackMetadata] = await Message.unpack(
+      encryptedMsg,
+      didResolver,
+      secretsResolver,
+      {},
+    );
+
+    const unpackedContent = unpackedMsg.as_value();
+    expect(unpackedContent).toMatchObject({
+      id: MESSAGE_ID,
+      typ: MESSAGE_TYP,
+      type: MESSAGE_TYPE,
+      body: MESSAGE_BODY,
+    });
+
+    expect(unpackMetadata).toMatchObject({
+      sign_from: expect.any(String),
+      encrypted_from_kid: ALICE_KID,
+      encrypted_to_kids: expect.any(Array),
+    });
+  });
+
+  it('should unpack an encrypted message successfully', async () => {
+    const msg = new Message({
+      id: MESSAGE_ID,
+      typ: MESSAGE_TYP,
+      type: MESSAGE_TYPE,
+      from: ALICE_DID,
+      to: [BOB_DID],
+      created_time: 1516269022,
+      expires_time: 1516385931,
+      body: MESSAGE_BODY,
+    });
+
+    // Pack the message to simulate sending it
+    const [encryptedMsg] = await msg.pack_encrypted(
+      BOB_DID,
+      ALICE_DID,
+      ALICE_DID,
+      didResolver,
+      secretsResolver,
+      {
+        forward: false,
+      },
+    );
+
+    // Unpack the message to simulate receiving it
+    didResolver = createDidResolver([ALICE_DID_DOC, BOB_DID_DOC]);
+    secretsResolver = createSecretsResolver(BOB_SECRETS);
+
+    const [unpackedMsg, unpackMetadata] = await Message.unpack(
+      encryptedMsg,
+      didResolver,
+      secretsResolver,
+      {},
+    );
+
+    expect(unpackedMsg).toBeDefined();
+    expect(unpackedMsg.as_value()).toMatchObject({
+      typ: MESSAGE_TYP,
+      body: MESSAGE_BODY,
+    });
+
+    expect(unpackMetadata).toMatchObject({
+      sign_from: expect.any(String),
```

Comment:
I think the "any" keyword  should be remove in here, its not allowed by the @typescript-eslint/no-explicit-any rule. I see its applied everywhere in the test, please remove it to avoid prudent errors like the one below

![Screenshot from 2024-11-05 09-01-45](https://github.com/user-attachments/assets/694532e3-0b6f-43a8-b046-619106c46ed9)

Reply:
The errors in the npm run lint (curiously, they don't come up in our pipeline! we must check that enforcement: https://github.com/ADORSYS-GIS/wallet-vc-libs/issues/39)

Regarding the errors in your screenshot, I added a line of: 
    // eslint-disable-next-line @typescript-eslint/no-explicit-any

To disable the alert since this MR does not touch those libraries and the change sounds too risky for very little (find out which datatype we want).

About the use of any in the use case, I find that it is still appropriate since we don't require more specific assertions.

Reply:
Actually the lint error is just on the dist folder, we don't care about lint there... closing the issue I created above and just leaving this comment like this.

Reply:
Oh @hugoib !
Sorry the issue is not even coming from you code, i didn't clearly read the logs.